pkg/btfhelpers/btfhelpers.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package btfhelpers provides a couple of helper functions to bridge Go's reflection system with
+// types from BTF
+package btfhelpers
+
+import (
+	"reflect"
+
+	"github.com/cilium/ebpf/btf"
+)
+
+func GetType(typ btf.Type) (reflect.Type, []string) {
+	switch typedMember := typ.(type) {
+	case *btf.Array:
+		arrType, typeNames := getSimpleType(typedMember.Type)
+		if arrType == nil {
+			return nil, nil
+		}
+		return reflect.ArrayOf(int(typedMember.Nelems), arrType), append(typeNames, typ.TypeName())
+	default:
+		arrType, typeNames := getSimpleType(typ)
+		return arrType, append(typeNames, typ.TypeName())
+	}
+}
+
+func GetUnderlyingType(tf *btf.Typedef) (btf.Type, []string, error) {
```

Comment:
Can you document the return values?

It seems it can never return an error. Could you remove `error`?